shared-libs/server-checks/src/checks.js

```diff
@@ -59,7 +60,43 @@ const couchDbNoAdminPartyModeCheck = () => {
       } else {
         console.error('Expected a 401 when accessing db without authentication.');
         console.error(`Instead we got a ${statusCode}`);
-        reject(new Error('CouchDB security seems to be misconfigured, see: https://github.com/medic/cht-core/blob/master/DEVELOPMENT.md#enabling-a-secure-couchdb'));
+        reject(new Error('CouchDB security seems to be misconfigured, ' +
+          'see: https://github.com/medic/cht-core/blob/master/DEVELOPMENT.md#enabling-a-secure-couchdb'));
+      }
+    }).on('error', (e) => {
+      reject(`CouchDB doesn't seem to be running on ${url.format(noAuthUrl)}. ` +
+        `Tried to connect but got an error:\n ${e.stack}`);
+    });
+  });
+};
+
+// since we don't know what the object keys are, so lets find the first one
+// and then the first element of that array. We always work in a single node
+//  membership, so this should be safe.
+const getNodeNameFromJson = (json) => {
+  if (typeof json === 'object' && Object.keys(json).length > 0) {
+    const key = Object.keys(json).shift();
+    if(Array.isArray(json[key]) && json[key].length > 0) {
+      return json[key].shift();
+    }
+  }
+  return false;
+};
```

Comment:
It's not necessary to be quite so defensive with internal functions - it's safe to assume `json` is an object. Also `json` isn't a very good variable name as it doesn't describe anything about what's _in_ the json. To be pedantic about the nomenclature for a second, technically it's not JSON because JSON is an object notation in String form - this is just an object.

Secondly we know the `_membership` endpoint will return `all_nodes` and `cluster_nodes` as keys, but we can't assume that `all_nodes` will be first, so just taking the first key isn't reliable.

Finally we should make sure this works with clustered configurations because we definitely want to support that some time, and this may block startup completely in these scenarios. It'd be better to err on the side of allowing the server to start at the risk of allowing some misconfigurations through.

Given the above, this is how I would write this function...

```suggestion
const checkNodeName = (nodeName, membership) => {
  return membership &&
    membership.all_nodes &&
    membership.all_nodes.includes(nodeName);
};
```

This is a little defensive in that it allows for null and empty responses, but assumes the CouchDB API is respected.

Reply:
Ah yes - thanks for the suggestion! Fixed in 76337dd.

I'd wanted to ask just how defensive we should be.  A lot of this stemmed from not knowing exactly which server variable we needed to match on. `all_nodes` it is!  While I did like knowing what was in the server, giving the error with the endpoint they can check themselves is good enough.